src/tsickle.ts

```diff
@@ -50,6 +50,11 @@ export interface AnnotatorHost {
    * If true, do not modify quotes around property accessors.
    */
   disableAutoQuoting?: boolean;
+  /**
+   * Annotator has to emit "exports.Foo" style statements if and only if also doing the googmodule
+   * emit transformation.
```

Comment:
Why?  Is this just avoiding the bug, or is there some fundamental reason?

(Also maybe this comment should open with something like "true if we want googmodule output" or some such)

Reply:
You mean as opposed to just emitting an `export {X};`?

The problem with just emitting `export {X};` is if `X` is a type, not a value, TypeScript will not convert it to an `exports.X` statement. If we make `X` a value, we're causing all kinds of problems with newly introduced type/value conflicts, and also the problem with the import/local var/export conflicts leading to a duplicate import.

Basically what we're doing here is instead of introducing a new local value symbol and having TS create the export for it, we're doing the `exports.` emit ourselves to avoid having to create the value symbol.

The long term for this code btw is to move to `googmodule`, but for that we'll have to figure out how to transport the `symbolToAliasMap` all the way over there.

Reply:
I think I don't understand the comment, it's saying one of
a) has to emit exports.Foo rather than nothing?
b) has to emit exports.Foo rather than export {Foo}?

and also I don't understand why it's conditioned on googmodule

Reply:
You're right. Essentially we cannot just stay with an ES6 export, because TS drops those. We can fix that by introducing a local variable (which would get exported), but we cannot do that because of the problems I'm fixing here.

So what remains is emitting an export "manually". I've keyed that on `googmodule`, but I think the better thing to do is keying it on whether we'll emit `CommonJS`. If we don't do that, we don't know how to emit the faux export, so we should not do it - but also, if we're not doing `CommonJS`, the downstream user cannot use `googmodule`, which means they cannot import the type from Closure anyway, so there's no point in doing this at all.